src/clj/puppetlabs/services/jruby/jruby_puppet_core.clj

```diff
@@ -206,6 +206,39 @@
                                                                             (:ruby-load-path jruby-config)))]
     (jruby-core/initialize-config (assoc modified-jruby-config :lifecycle lifecycle-fns))))
 
+(schema/defn initialize-configs-and-create-pool-context
+  "Handles initializing the jruby and jruby-puppet configs, and then uses them
+  to create a pool-context"
```

Comment:
This looks great! I think the name and this docstring should be changed though, because the function isn't creating the pool context anymore.